### ISSUE TITLE
Add way to do cross check implementations

### ISSUE THREAD

**Issue by zhoonit** [nnstreamer/nntrainer#133]
Currently there is limited way to check if the current implementation is logically correct. (Aside from it is running and have pretty good result.)

Cross checking the implementation with other frameworks is needed.

**Comment by kparichay** [comment]
Make tensorflow operations for each layer:
1. Forward output
2. Gradient for the weights in layer
3. Updated weight

Networks to use:
1. 1 FC layer with varied activations and loss training - tests FC layer, activations and loss
2. 2 FC layers with training - tests transfer of gradient from back from FC layer

Add more for other layers.

**Comment by jijoongmoon** [comment]
We checked bullet items as below with tensorflow.
. conv2d : done
. fc : done
. pooling2d : done
. flatten : done
. optimizer : sgd done
. loss : mse, cross-sigmoid, cross-softmax : done

**Comment by kparichay** [comment]
@jijoongmoon conv2d backward verification is still to be done as per https://github.com/nnstreamer/nntrainer/blob/master/test/unittest/unittest_nntrainer_layers.cpp#L949

**Comment by jijoongmoon** [comment]
Adam validation is done with #376

**Comment by zhoonit** [comment]
Can this be closed now?

**Comment by kparichay** [comment]
Once MNIST comparison with tensorflow is done, add this to a test which is run in every PR.

**Comment by jijoongmoon** [comment]
We are close this issues. We reopen or make new issue after implementing Transpose convolution.
